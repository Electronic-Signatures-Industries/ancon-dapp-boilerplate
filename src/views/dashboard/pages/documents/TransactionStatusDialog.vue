<template>
  <v-dialog
    v-model="show"
    max-width="500px"
  >
    <v-card>
      <v-card-title>
        <span class="headline">Transacci√≥n</span>
      </v-card-title>

      <v-card-text>
        {{ transactionStatus }}
      </v-card-text>

      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn
          color="blue darken-1"
          text
          @click="setVisible(false)"
        >
          Cancel
        </v-btn>

        <v-btn
          data-cy="confirm"
          color="blue darken-1"
          text
          @click="onConfirm()"
        >
          OK
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script lang="ts">
import Vue from 'vue'
import Component from 'vue-class-component';
import { Emit, Prop } from 'vue-property-decorator';

@Component({})
export default class TransactionStatusDialog extends Vue {
  @Prop()
  readonly show: boolean;

  @Prop()
  readonly transactionStatus?: string;

  @Emit('update:show')
  setVisible(newStatus: boolean) {
    return newStatus;
  }

  @Emit('confirm')
  onConfirm() {}
} 
</script>