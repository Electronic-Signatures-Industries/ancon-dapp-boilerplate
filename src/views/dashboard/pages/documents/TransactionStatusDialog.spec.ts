import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";
import TransactionStatusDialog from "./TransactionStatusDialog.vue";

describe("TransactionStatusDialog.vue", () => {
  const resultMock = jest.fn<void, []>();
  let wrapperComponent: Wrapper<Vue>;
  let component: Wrapper<TransactionStatusDialog>;

  beforeEach(async () => {
    const localVue = createLocalVue();

    const App = localVue.component("App", {
      components: { TransactionStatusDialog },
      data() {
        return { show: true };
      },
      methods: {
        result: resultMock,
      },
      template: `
        <v-app>
          <transaction-status-dialog
            id="dialog"
            :show.sync="show"
            :transactionStatus="\'Lorem Ipsum\'"
            @confirm="result"
          />
        </v-app>
      `,
    });

    wrapperComponent = mount(App, {
      localVue,
      vuetify: new Vuetify(),
      attachToDocument: true,
    });

    component = wrapperComponent.find("#dialog");
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should match snapshot", () => {
    expect(wrapperComponent).toMatchSnapshot();
  });

  it("should send the 'confirm' event when the OK button is pressed", async () => {
    const button = component.find("[data-cy=confirm]");
    button.trigger("click");
    expect(resultMock).toHaveBeenCalledTimes(1);
  });
});
