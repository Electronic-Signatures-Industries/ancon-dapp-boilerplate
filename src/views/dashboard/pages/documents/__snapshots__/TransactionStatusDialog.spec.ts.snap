// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionStatusDialog.vue should match snapshot 1`] = `
<div data-app="true" class="v-application v-application--is-ltr theme--light" id="app">
  <div class="v-overlay theme--dark" style="z-index: 5;">
    <div class="v-overlay__scrim" style="opacity: 0; background-color: rgb(33, 33, 33); border-color: #212121;"></div>
  </div>
  <div class="v-application--wrap">
    <div role="dialog" class="v-dialog__container" id="dialog"></div>
  </div>
</div>
`;
